eng/illink.targets

```diff
@@ -1,4 +1,11 @@
 <Project>
+  <PropertyGroup>
+    <PrepareResourcesDependsOn>
+      _EmbedILLinkSubstitutionsXmls;
```

Comment:
Just wondering, does ordering matter here? If so we should probably add a comment above in case somebody decides to either move this property group somewhere else, or move the illink.targets import sooner/later.

Reply:
> does ordering matter here?

It's MSBuild, the ordering always matters. 😉 

> in case somebody decides to either move this property group somewhere else

I'm not sure I understand the concern here. Why would someone move this property group somewhere else? It is hooking the `_EmbedILLinkSubstitutionsXmls` target, which is defined in this file, to run before the `PrepareResources` target. Someone splitting the target, and how the target gets hooked up in the build would be a mistake.

> or move the illink.targets import sooner/later.

Where/when the illink.targets import occurs won't matter. As long as someone isn't overwriting the `$(PrepareResourcesDependsOn)` property (which would be bad for many reasons), this approach will always work no matter when the .targets is imported. The `_EmbedILLinkSubstitutionsXmls` target will run before `PrepareResources` is run.

Reply:
What I meant is if we required your target to run before other PrepareResourcesDependsOn since depending on when this file is imported, this property might get redefined in later to something like:
```xml
    <PrepareResourcesDependsOn>
      SomeOtherTarget;
      $(PrepareResourcesDependsOn)
    </PrepareResourcesDependsOn>
```

Which will cause your target to not run before the rest of the PrepareREsourcesDependsOn targets. From the looks of it this is not the case, as it looks like you really only want to run before PrepareResources target and if so this is the right way, I was just saying that if for whatever reason you also wanted to make sure you ran before the rest of the dependsOn targets, we might want to add a comment to point out that ordering of that property matters here.

Reply:
> if we required your target to run before other PrepareResourcesDependsOn 

Ah, got it. If that was the case, I think it would be better to use a `BeforeTargets` instead. But here I don't care about the ordering of any other `PrepareResourcesDependsOn` targets. I just need my target to run before `PrepareResources` because that is the target that consumes `EmbeddedResource`, which is what we are adding the combined xml file to.